### PR TITLE
Added MapBasedParser and SourceParser also test it

### PR DESCRIPTION
**Pull request by ikfir**

### CODE REVIEW

File: commands/src/main/java/net/apartium/cocoabeans/commands/parsers/MapBasedParser.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    public Optional<ParseResult<T>> parse(CommandProcessingContext commandProcessingContext) {
+        List<String> args = commandProcessingContext.args();
+        int index = commandProcessingContext.index();
+
+        Map<String, T> map = getMap();
+        String s = "";
+        for (int i = index; i < args.size(); i++) {
+            if (i != index)
+                s += " ";
+            s += args.get(index);
```
Comment (liorsl) [inline commands/src/main/java/net/apartium/cocoabeans/commands/parsers/MapBasedParser.java:26]: This is a very weird behavior that can lead to a lot of confusion, are you sure this is needed?

File: commands/src/main/java/net/apartium/cocoabeans/commands/parsers/SourceParser.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
+package net.apartium.cocoabeans.commands.parsers;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
```
Comment (liorsl) [inline commands/src/main/java/net/apartium/cocoabeans/commands/parsers/SourceParser.java:7]: Add documentation for this thing

File: commands/src/test/java/net/apartium/cocoabeans/commands/ThrowItCommand.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
+package net.apartium.cocoabeans.commands;
+
+import net.apartium.cocoabeans.commands.parsers.SourceParser;
+
+import java.util.Map;
+
+@Command("throw-it")
+public class ThrowItCommand implements CommandNode {
```
Comment (liorsl) [inline commands/src/test/java/net/apartium/cocoabeans/commands/ThrowItCommand.java:8]: Rename to CommandWithInvalidSourceParser?

**Review by liorsl — CHANGES_REQUESTED** [review]

File: commands/src/main/java/net/apartium/cocoabeans/commands/parsers/MapBasedParser.java (hunk 1 above)
Comment (ikfir) [reply commands/src/main/java/net/apartium/cocoabeans/commands/parsers/MapBasedParser.java:26]: yeah if you have

Map<String, ?> map = Map.of("number ten", 10, "boolean false", false);

**Review by liorsl — APPROVED** [review]

**Pull request merged**